From 607d76c166402e78aa0f8b38a39fda73fc0d55b7 Mon Sep 17 00:00:00 2001
From: Bin Han <bin.b.han@intel.com>
Date: Tue, 16 Sep 2014 14:23:42 +0800
Subject: [PATCH 144/164] atomisp2: add per_frame_setting flag for
 atomisp_parameters

BZ: 223442

In the past, HAL uses the isp_config_id == 0 to indicate this is a global
setting, but this is not true with a new change request, that HAL wants to set
>0 isp_config_id for global setting. In such way, HAL can know the 3A
statistics comes from which parameter.

Therefore we add a new flag per_frame_setting to indicate if this setting is
for global or per_frame.

Change-Id: Idabf6390782a83d2317782217c3a270042c14fda
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-8929
Signed-off-by: Bin Han <bin.b.han@intel.com>
Reviewed-on: https://android.intel.com:443/260891
---
 include/linux/atomisp.h |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/include/linux/atomisp.h b/include/linux/atomisp.h
index 100c97f..142b565 100644
--- a/include/linux/atomisp.h
+++ b/include/linux/atomisp.h
@@ -552,6 +552,14 @@ struct atomisp_parameters {
 	 * frame, driver will send this id back with output frame together.
 	 */
 	uint32_t	isp_config_id;
+
+	/*
+	 * Switch to control per_frame setting:
+	 * 0: this is a global setting
+	 * 1: this is a per_frame setting
+	 * PLEASE KEEP THIS AT THE END OF THE STRUCTURE!!
+	 */
+	uint32_t	per_frame_setting;
 };
 
 #define ATOMISP_GAMMA_TABLE_SIZE        1024
-- 
1.7.9.5

